### ISSUE TITLE
simplefilter decorator does not work as documented

### ISSUE THREAD

**Issue by Anteru** [pygments/pygments#782]
(Original issue 1075 created by alexander_belopolsky on 2015-01-19T17:51:25.958648+00:00)

When I create a filter as described in the  ["Write your own filter"](http://pygments.org/docs/filterdevelopment/) document:

```
@simplefilter
def myfilter(lexer, stream, options):
      ..

lexer.add_filter(myfilter())
```

I get

```
TypeError: myfilter() takes 3 positional arguments but 4 were given
```

I have to work around by adding an extra argument to `myfilter`:

```
@simplefilter
def myfilter(self, lexer, stream, options):
      ..
```

I am using

```
$ pygmentize -V
Pygments version 2.0.1, (c) 2006-2014 by Georg Brandl.
```

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2015-01-21T06:26:22.339447+00:00)

Closes #1075: fix docs for @simplefilter.

The function needs a "self" argument since it is used as a method in the
automatically created class.  No use fixing that in the code, as it
kills backwards compatibility.

→ <<cset 8f531b593818>>

**Comment by Anteru** [comment]
(Original comment by birkenfeld on 2015-01-21T06:26:22.387896+00:00)

Closes #1075: fix docs for @simplefilter.

The function needs a "self" argument since it is used as a method in the
automatically created class.  No use fixing that in the code, as it
kills backwards compatibility.

→ <<cset c98934fb123b>>
